client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/MongoInfo.java

```diff
@@ -0,0 +1,29 @@
+package org.evomaster.client.java.instrumentation;
+
+import java.io.Serializable;
+
+/**
+ * Info related to MONGO command execution.
+ *
+ * This class is IMMUTABLE
```

Comment:
@hghianni this is not necessarily the case. you have references to `Object` type (eg, `collection`) that can be mutated from outside, making this class not immutable.

Either remove the comment about immutability, or enforce immutability (which might be tricky when dealing with `Object` refs, as would need to do deep-clones on input)